src/requirements/reqs-functions.ts

```diff
@@ -8,6 +9,121 @@ import {
   GroupedRequirementFulfillmentReport,
 } from './types';
 
+type FulfillmentStatistics = {
+  readonly id: string;
+  readonly requirement: RequirementWithIDSourceType;
+  readonly courses: readonly (readonly CourseTaken[])[];
+} & RequirementFulfillmentStatistics;
+
+/**
+ * @param course course object with useful information retrived from Cornell courses API.
+ * @returns true if the course is AP/IB equivalent course or credit
+ */
+const courseIsAPIB = (course: CourseTaken): boolean =>
+  [CREDITS_COURSE_ID, FWS_COURSE_ID].includes(course.courseId) ||
+  ['AP', 'IB', 'CREDITS'].includes(course.subject);
+
+/**
+ * Used for total academic credit requirements for all colleges except EN and AR
+ * @param course course object with useful information retrived from Cornell courses API.
+ * @returns true if the course is not PE or 10** level
+ */
+const courseIsAllEligible = (course: CourseTaken): boolean =>
+  course.courseId === CREDITS_COURSE_ID ||
+  (!courseIsAPIB(course) && course.subject !== 'PE' && !course.number.startsWith('10'));
+
+const getTotalCreditsFulfillmentStatistics = (
+  college: string,
+  courses: readonly CourseTaken[]
+): FulfillmentStatistics | null => {
+  const requirementCommon = {
+    sourceType: 'College',
+    sourceSpecificName: college,
+    name: 'Total Academic Credits',
```

Comment:
Would the `name` field be how the frontend would special case the 120 total academic credits?

Reply:
No, it's handled by completely remove them from the requirement json, and inject it as the first college requirement in the requirement menu.